d2/src/main/java/com/linkedin/d2/xds/XdsToD2PropertiesAdaptor.java

```diff
@@ -355,16 +352,16 @@ private String getSymlink(String actualNodeName) {
     }
   }
 
-  private static String removeNodePathPrefix(String path, String prefix)
+  private static String getNodeName(String path)
   {
-    int idx = path.indexOf(prefix);
+    int idx = path.lastIndexOf(PATH_SEPARATOR);
     if (idx == -1)
     {
       return path;
     }
     else
     {
```

Comment:
Kind of a nit, but since you're doing `idx + 1`, you don't need to check for `idx == -1` anymore. This can be simplified to:
```java
private static String getNodeName(String path)
{
  return path.substring(path.lastIndexOf(PATH_SEPARATOR) + 1);
}
```

Reply:
ah, yeah, updated.